monai/data/utils.py

```diff
@@ -76,8 +76,11 @@
     "pad_list_data_collate",
     "no_collation",
     "convert_tables_to_dicts",
+    "SUPPORTED_PICKLE_MOD",
 ]
 
+SUPPORTED_PICKLE_MOD = {"pickle": pickle}
```

Comment:
Thanks for the quick fix for the issue.
In this way, I feel we can't support any other pickle tool unless hard-code here first, other pickle tool should be available in PyTorch `torch.save`.

Thanks.

Reply:
Hi @Nic-Ma, I've checked that it is possible to edit the dictionary from the user side:
```py
>>> from monai.data.dataset import PersistentDataset
>>> PersistentDataset([1, 2, 3], lambda x: x, cache_dir=None, pickle_module="test")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Documents/MONAI/monai/data/dataset.py", line 185, in __init__
    self.pickle_module = look_up_option(pickle_module, SUPPORTED_PICKLE_MOD)
  File "/Users/Documents/MONAI/monai/utils/module.py", line 121, in look_up_option
    raise ValueError(f"Unsupported option '{opt_str}', " + supported_msg)
ValueError: Unsupported option 'test', Available options are {'pickle'}.

>>> import pickle
>>> from monai.data import utils
>>> utils.SUPPORTED_PICKLE_MOD["test"] = pickle
>>> PersistentDataset([1, 2, 3], lambda x: x, cache_dir=None, pickle_module="test")
<monai.data.dataset.PersistentDataset object at 0x7fd5b809fe50>

```